tpm2/structures.go

```diff
@@ -1028,3 +1029,70 @@ type TPMLDigest struct {
 	Count   uint32
 	Digests []tpmutil.U16Bytes
 }
+
+// Encode converts the TPMLDigest structure into a byte slice
+func (list *TPMLDigest) Encode() ([]byte, error) {
+	if uint32(len(list.Digests)) != list.Count {
+		return nil, fmt.Errorf("Count invalid. Have: %v - Want: %v", len(list.Digests), list.Count)
+	}
+	res, err := tpmutil.Pack(list.Count)
+	if err != nil {
+		return nil, err
+	}
+	for _, item := range list.Digests {
+		b, err := tpmutil.Pack(item)
+		if err != nil {
+			return nil, err
+		}
+		for _, i := range b {
+			res = append(res, i)
+		}
+	}
+	return res, nil
+}
+
+// DecodeTPMLDigest decodes a TPML_Digest part of a message. No error is returned if
+// the input has extra trailing data.
+func DecodeTPMLDigest(buf []byte) (*TPMLDigest, error) {
+	in := bytes.NewBuffer(buf)
+	var tpmld TPMLDigest
+	var err error
+	var hashlen uint16
+	if err = binary.Read(in, binary.BigEndian, &tpmld.Count); err != nil {
+		return &tpmld, fmt.Errorf("decoding TPML_Digest: %v", err)
```

Comment:
Should this function return nil instead of a partly initialized TPMLDigest if there is an error? (And again in a few other places in the current commit)